typescript/creational/builder/Readme.md

```diff
@@ -0,0 +1,2 @@
+## Builder Pattern
```

Comment:
The pattern itself for the concrete implementation doesn't need a readme so this can be removed.